website/docs/r/sentinel_automation_rule.html.markdown

```diff
@@ -74,12 +74,21 @@ The following arguments are supported:
 
 ~> **Note:** Either one `action_incident` block or `action_playbook` block has to be specified.
 
-* `condition` - (Optional) One or more `condition` blocks as defined below.
+* `condition` - (Optional / **Deprecated** ) One or more `condition` blocks as defined below.
+
+~> **Note:** `condition` only supports the [`Property` condition type](https://learn.microsoft.com/en-us/rest/api/securityinsights/preview/automation-rules/create-or-update?tabs=HTTP#propertyconditionproperties). Please use `condition_json` if you want other condition types.
+
+* `condition_json` - (Optional) A JSON array of one or more condition JSON objects as is defined [here](https://learn.microsoft.com/en-us/rest/api/securityinsights/preview/automation-rules/create-or-update?tabs=HTTP#automationruletriggeringlogic).
```

Comment:
Instead of replacing this with json to cover the types, can we please add 3 different block types like `condition_boolean`, `condition_property` etc

Reply:
That is my first attempt, while later I noticed the condition can be nested in multiple depths, which we can't define a schema to support.

Reply:
I'm not sure what you mean? given that there are different types here we should be able to create different blocks for each one?

Reply:
What I meant is that the `condition` here is a polymorphism, which can be things like `condition{Boolean}`, `condition{PropertyChanged}`, etc. While, each such variant can contain property called `innerConditions`, which is an array of `condition`. That means condition as below is a valid one:

```
  "conditions": [
    {
      "conditionType": "Boolean",  # NOTE: This is a boolean condition
      "conditionProperties": {
        "operator": "Or",
        "innerConditions": [
          {
            "conditionType": "Property",
            "conditionProperties": {
              "propertyName": "IncidentTitle",
              "operator": "Equals",
              "propertyValues": [
                "a"
              ]
            }
          },
          {
            "conditionType": "Boolean",  # NOTE: This is a boolean condition, but is nested in another boolean condition
            "conditionProperties": {
              "operator": "And",
              "innerConditions": [
                {
                  "conditionType": "Property",
                  "conditionProperties": {
                    "propertyName": "IncidentTitle",
                    "operator": "Equals",
                    "propertyValues": [
                      "b"
                    ]
                  }
                },
                {
                  "conditionType": "Property",
                  "conditionProperties": {
                    "propertyName": "IncidentTitle",
                    "operator": "Equals",
                    "propertyValues": [
                      "c"
                    ]
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
```

Reply:
So shouldn't it just be a condition_bool condition_property each with that specific type?

checking the SDK is not just a single property taking json, we should be providing fully realized terraform blocks to match it not just a json blob? which is not very user friendly or providing a good UX

Reply:
The SDK model can afford recursive embeddings, while I don't think Terraform schema can afford to that?

Reply:
ahh thanks for explaining, that makes sense then!